Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -302,12 +354,22 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
     
     /// The documentation bundles that are currently registered with the context.
     public var registeredBundles: some Collection<DocumentationBundle> {
-        return dataProvider.bundles.values
+        switch dataProvider {
+        case .legacy(let legacyDataProvider):
+            Array(legacyDataProvider.bundles.values)
+        case .new:
+            bundle.map { [$0] } ?? []
+        }
     }
     
     /// Returns the `DocumentationBundle` with the given `identifier` if it's registered with the context, otherwise `nil`.
     public func bundle(identifier: String) -> DocumentationBundle? {
-        return dataProvider.bundles[identifier]
+        switch dataProvider {
+        case .legacy(let legacyDataProvider):
+            legacyDataProvider.bundles[identifier]
+        case .new:
+            bundle?.identifier == identifier ? bundle : nil
```

Comment:
Should we have an assertion on this line? ...rather than just returning `nil`? Likely if there's a mismatch in the single bundle's identifier there's some coding mistake elsewhere.

Reply:
We could, since it would be a coding mistake to pass another bundle identifier to this method. 

This method is [deprecated](https://github.com/swiftlang/swift-docc/pull/1059/files#diff-c452db10128fc43b2e77212a1a282fe0834e541e10852fa8dd50f7c9b480b4e7R372) in #1059 so I don't think it's _too_ important to assert here since there would already be deprecation warnings for any code that calls this, but if I'm already adding an assertion to `contentsOfURL(_:in:)` then I can add it here as well.

Reply:
Added in https://github.com/swiftlang/swift-docc/pull/1057/commits/96065616aec25b87597cb0a14790d3ea5d9d890d